### PR TITLE
audit create - infrastructure

### PR DESCRIPTION
**Pull request by AyalK**
SEARCH-54894

### CODE REVIEW

File: main/src/main/java/com/kenshoo/template/main/AuditGroupTable.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import java.sql.Timestamp;
+
+public class AuditGroupTable extends TableImpl<Record> {
+
+    public static final AuditGroupTable TABLE = new AuditGroupTable("audit_groups");
+
+    public final TableField<Record, Long> id = createField("id", SQLDataType.BIGINT);
+    public final TableField<Record, String> name = createField("name", SQLDataType.VARCHAR.length(255));
+    public final TableField<Record, String> clientId = createField("client_id", SQLDataType.VARCHAR.length(255));
+    public final TableField<Record, String> status = createField("status", SQLDataType.VARCHAR.length(255));
```
Comment (nirfeldman) [inline main/src/main/java/com/kenshoo/template/main/AuditGroupTable.java]: Did you consider using an enum type in the DB
A varchar 256 sounds a bit too much for a status

File: main/src/main/java/com/kenshoo/template/main/AuditGroupDao.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    public long create(AuditGroupEntity auditGroupEntity) {
+        dslContext.insertInto(TABLE)
+                .set(TABLE.id, auditGroupEntity.getId())
+                .set(TABLE.name, auditGroupEntity.getName())
+                .set(TABLE.clientId, auditGroupEntity.getClientId())
+                .set(TABLE.status, auditGroupEntity.getStatus())
+                .set(TABLE.creationDate, auditGroupEntity.getCreationDate())
+                .execute();
+        return dslContext.lastID().longValue();
```
Comment (SharonHart) [inline main/src/main/java/com/kenshoo/template/main/AuditGroupDao.java]: @yuval-rikover 
The return of lastID :)
@AyalK I will explain tomorrow

File: main/src/main/java/com/kenshoo/template/main/AuditGroupDao.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    public int update(AuditGroupEntity auditGroupEntity) {
```
Comment (SharonHart) [inline main/src/main/java/com/kenshoo/template/main/AuditGroupDao.java]: Let's not add unneeded methods, for now,  concentrating on creation.

File: main/src/main/java/com/kenshoo/template/main/AuditMembershipEntity.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
+package com.kenshoo.template.main;
+
+import java.util.Objects;
+
+public class AuditMembershipEntity {
```
Comment (SharonHart) [inline main/src/main/java/com/kenshoo/template/main/AuditMembershipEntity.java]: I would name it just AuditMembership, its name and content hints enough on that it's an entity.

File: main/src/main/java/com/kenshoo/template/main/AuditRuleDao.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        dslContext.insertInto(TABLE)
+                .set(TABLE.id, auditRuleEntity.getId())
+                .set(TABLE.auditGroupId, auditRuleEntity.getAuditGroupId())
+                .set(TABLE.viewName, auditRuleEntity.getViewName())
+                .set(TABLE.status, auditRuleEntity.getStatus())
+                .execute();
+        return dslContext.lastID().longValue();
+    }
+
+    public int update(AuditRuleEntity auditRuleEntity) {
```
Comment (SharonHart) [inline main/src/main/java/com/kenshoo/template/main/AuditRuleDao.java]: not in this scope

File: main/src/main/java/com/kenshoo/template/main/AuditRuleDao.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                .set(TABLE.id, auditRuleEntity.getId())
+                .set(TABLE.auditGroupId, auditRuleEntity.getAuditGroupId())
+                .set(TABLE.viewName, auditRuleEntity.getViewName())
+                .set(TABLE.status, auditRuleEntity.getStatus())
+                .where(TABLE.id.eq(auditRuleEntity.getId()))
+                .execute();
+    }
+
+
+    public int updateStatus(long id, String status) {
```
Comment (SharonHart) [inline main/src/main/java/com/kenshoo/template/main/AuditRuleDao.java]: not in this scope

File: main/src/main/java/com/kenshoo/template/main/AuditRuleDao.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        return dslContext.update(TABLE)
+                .set(TABLE.status, status.toUpperCase())
+                .where(TABLE.id.eq(id))
+                .execute();
+    }
+
+    public AuditRuleEntity getAuditRuleEntity(long id) {
```
Comment (SharonHart) [inline main/src/main/java/com/kenshoo/template/main/AuditRuleDao.java]: not in this scope

File: main/src/main/java/com/kenshoo/template/main/AuditGroupEntity.java [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+
+        return Objects.hash(id, name, clientId, status, creationDate);
+    }
+
+    public static class Builder {
+
```
Comment (SharonHart) [inline main/src/main/java/com/kenshoo/template/main/AuditGroupEntity.java]: have a private constructor to hide from the user.
private Builder() { }

File: main/src/main/java/com/kenshoo/template/main/AuditMembershipEntity.java (hunk 4 above)
Comment (SharonHart) [reply main/src/main/java/com/kenshoo/template/main/AuditMembershipEntity.java]: For the other classes as well

File: main/src/main/java/com/kenshoo/template/main/AuditGroupTable.java (hunk 1 above)
Comment (yuval-rikover) [reply main/src/main/java/com/kenshoo/template/main/AuditGroupTable.java]: I think using varchar is more convenient; enumType means that modifying the enum requires Alter table.
Also, we use "manual" jooq queries (not JPA managed entities/daos), so the added convenience is missed out, and you have to use EnumConverter

**Comment by yuval-rikover** [comment]
@AyalK remove unused import 
import java.util.Objects;

**Pull request merged**